src/ilcdlib/epd/dialect/epddenmark.py

```diff
@@ -0,0 +1,57 @@
+#
+#  Copyright 2024 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+from typing import Any
+
+from ilcdlib.dto import OpenEpdIlcdOrg
+from ilcdlib.entity.contact import IlcdContactReader
+from ilcdlib.entity.flow import UriBasedIlcdFlowReader
+from ilcdlib.epd.reader import IlcdEpdReader
+from ilcdlib.type import LangDef
+
+
+class EpdDenmarkIlcdXmlEpdReader(IlcdEpdReader):
+    """Reader for EPDs in the Denmark specific ILCD XML format."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs, flow_reader_cls=UriBasedIlcdFlowReader)
+
+    @classmethod
+    def is_known_url(cls, url: str) -> bool:
+        """Return whether the URL recognized as a known Denmark URL."""
+        return "ecosmdp" in url.lower()
```

Comment:
I'm afraid we can't rely on this. ecosmdp hosts a lot of different data. This method should always return False and autodetection of the dialect won't work for EPD Denmark. (unless you know a better way to detect)

Reply:
Totally agree, updated